### PR TITLE
Use install target in ChainerX setup to install C++ headers too

### LINKED ISSUE #7422: Should install ChainerX C++ headers with libchainerx.so on pip install
Currently most easiest way to install ChainerX is to build via pip but pip doesn't install headers.
So it should install C++ headers too for convenience.

### PR DESCRIPTION
**Pull request by take-cheeze**
closes #7422

### CODE REVIEW

File: chainerx_build_helper.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def build_extension(self, ext):
             os.path.dirname(self.get_ext_fullpath(ext.name)))
         cmake_args = [
             '-DCHAINERX_BUILD_PYTHON=1',
-            '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + extdir,
```
Comment (hvy) [inline chainerx_build_helper.py]: Should we not keep `CMAKE_LIBRARY_OUTPUT_DIRECTORY` defined?

File: chainerx_build_helper.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def build_extension(self, ext):
...
+            '-DCMAKE_INSTALL_PREFIX=' + extdir,
```
Comment (hvy) [inline chainerx_build_helper.py]: Installing to `extdir` is fine for you purpose? I mean, is it okay that it's not configurable?

Comment (take-cheeze) [reply chainerx_build_helper.py]: When I need to make this configurable I'll run cmake without `pip` so this is what I expect

File: chainerx_build_helper.py (hunk 1 above)
Comment (take-cheeze) [reply chainerx_build_helper.py]: Defining `CMAKE_LIBRARY_OUTPUT_DIRECTORY` seems a bad hack to me.
Built library should be placed to build directory until it's installed.

What I want is a C++ headers corresponding to the `libchainerx.so` so the headers should be installed with the libraries so I'm using the cmake's install feature.

**Comment by hvy** [comment]
Jenkins, test this please.

**Pull request closed without merging**